Rebound: add rev2 and thus rev1 as well

Rossman360 opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

I made a rev2 for Rebound, a 49-key version. Then, I made rev1 files for the 48-key version and created default keymaps for each

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

<!--- Describe your changes in detail here. -->

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

drashna commented on keyboards/montsinger/rebound/keymaps/default49/keymap.c:
```diff
@@ -0,0 +1,163 @@
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+enum layer_names
+{
```
Just a nit pick: 
```suggestion
enum layer_names {
```

drashna commented on keyboards/montsinger/rebound/rev2/rules.mk:
```diff
@@ -0,0 +1,32 @@
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
+
+LAYOUTS = ortho_5x12
+LAYOUTS_HAS_RGB=no
```
```suggestion
LAYOUTS_HAS_RGB = no
```

drashna commented:
Okay, taking a closer look at this, both revisions lack a "default" keymap. 

Additionally, the revision 2 errors out with the default community layout (too many keys).   And it looks like it doesn't actually match the 5x12 or 4x12 ortho layouts.    And if that is the case, support (and naming) for it should be removed.

Rossman360 commented:
"buncha stuff" PR

-renamed ortho_5x12 to "49" and ortho_4x12 to "48" to reflect the number of keys rather than how the matrix is set up.
-made rev2 default
-added rotary pins for rev2
-renamed default49 keymap to default
-set up rev1 so it doesn't fail with 49-key keymap (still unsure how to set a default keymap for a specific rev)

zvecr commented on keyboards/montsinger/rebound/rev2/rev2.h:
```diff
@@ -0,0 +1,40 @@
+ * represents the switch matrix.
+ */
+ 
+#define LAYOUT_ortho_5x12( \
```
This is not `ortho_5x12 compatible
```suggestion
#define LAYOUT( \
```

zvecr commented on keyboards/montsinger/rebound/rev2/rules.mk:
```diff
@@ -0,0 +1,33 @@
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
+
+LAYOUTS = ortho_5x12
+LAYOUTS_HAS_RGB = no
```
```suggestion
```

zvecr commented on keyboards/montsinger/rebound/rules.mk:
```diff
@@ -28,5 +28,7 @@ AUDIO_ENABLE = no           # Audio output on port C6
 
-LAYOUTS = ortho_4x12
-LAYOUTS_HAS_RGB=no
+LAYOUTS = ortho_5x12
+LAYOUTS_HAS_RGB = no
+
```
```suggestion
```

zvecr commented on keyboards/montsinger/rebound/rev1/rules.mk:
```diff
@@ -0,0 +1,32 @@
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
+
+LAYOUTS = 48
+LAYOUTS_HAS_RGB=no
```
```suggestion

LAYOUTS = ortho_4x12
```

zvecr commented on keyboards/montsinger/rebound/rev1/rev1.h:
```diff
@@ -0,0 +1,39 @@
+ * represents the switch matrix.
+ */
+ 
+#define LAYOUT_48( \
```
```suggestion
#define LAYOUT_ortho_4x12 \
```

Rossman360 commented:
hang on, I'm changing everything

zvecr commented on keyboards/montsinger/rebound/rebound.h:
```diff
@@ -37,3 +43,16 @@
   { K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B } \
 }
+
+#define LAYOUT_49( \
```
Is this compatible with both rev1 and rev2?

zvecr commented on keyboards/montsinger/rebound/rebound.h:
```diff
@@ -26,7 +32,7 @@
  */
  
-#define LAYOUT_ortho_4x12( \
+#define LAYOUT_48( \
```
Is this compatible with both rev1 and rev2?

Also, this should still be ortho_4x12 based
```suggestion
#define LAYOUT_ortho_4x12( \
```

zvecr commented on keyboards/montsinger/rebound/rebound.h:
```diff
@@ -14,6 +14,12 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#ifdef KEYBOARD_montsinger_rebound_rev1
+    #include "rev1.h"
+#elif KEYBOARD_montsinger_rebound_rev2
+    #include "rev2.h"
+#endif
+
 #pragma once 
 
 #include "quantum.h"
```
```suggestion
#pragma once 
#include "quantum.h"

#ifdef KEYBOARD_montsinger_rebound_rev1
    #include "rev1.h"
#elif KEYBOARD_montsinger_rebound_rev2
    #include "rev2.h"
#8397 
```

zvecr commented on keyboards/montsinger/rebound/keymaps/rossman360/keymap.c:
```diff
@@ -16,40 +16,42 @@ enum layer_names {
-    JUMPBACK, KC_A   , KC_S   , KC_D   , KC_F   , KC_G   ,   KC_H   , KC_J   , KC_K   , KC_L   , KC_SCLN, KC_QUOT,
-    KC_LSFT , KC_Z   , KC_X   , KC_C   , KC_V   , KC_B   ,   KC_N   , KC_M   , KC_COMM, KC_DOT , KC_SLSH, KC_MINS,
-    CMDBSP  , ALTDEL , KC_NO  , CTRLSP , TABMOD , PGMOD  ,   ENTMOD , SPCMOD , MO(_DEL),KC_NO  , KC_BSPC, KC_DEL
+[_BASE] = LAYOUT_ortho_5x12(
```
None of these add an additional row to match the layout macro

Rossman360 commented:
I redid everything so that there is only one rev, added an info.json with both available layouts, renamed the layouts to LAYOUT (49 key) and LAYOUT_48. The default48 keymap disables the rotary encoder and utilizes the appropriate layout

zvecr commented on keyboards/montsinger/rebound/rules.mk:
```diff
@@ -28,5 +28,5 @@ AUDIO_ENABLE = no           # Audio output on port C6
 
-LAYOUTS = ortho_4x12
-LAYOUTS_HAS_RGB=no
+LAYOUTS_HAS_RGB = no
```
```suggestion
LAYOUTS = ortho_4x12
LAYOUTS_HAS_RGB = no
```

zvecr commented on keyboards/montsinger/rebound/info.json:
```diff
@@ -0,0 +1,127 @@
+                {"x":12, "y":4}
+            ]
+        },
+        "LAYOUT_48": {
```
```suggestion
        "LAYOUT_ortho_4x12": {
```

zvecr commented on keyboards/montsinger/rebound/keymaps/default48/keymap.c:
```diff
@@ -0,0 +1,163 @@
+ * |Adjust| Ctrl | Alt  | GUI  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = LAYOUT_48(
```
```suggestion
[_QWERTY] = LAYOUT_ortho_4x12(
```

zvecr commented on keyboards/montsinger/rebound/keymaps/default48/keymap.c:
```diff
@@ -0,0 +1,163 @@
+ * |Adjust| Ctrl | Alt  | GUI  |Lower |Space |Space |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_COLEMAK] = LAYOUT_48(
```
```suggestion
[_COLEMAK] = LAYOUT_ortho_4x12(
```

zvecr commented on keyboards/montsinger/rebound/keymaps/default48/keymap.c:
```diff
@@ -0,0 +1,163 @@
+ * |Adjust| Ctrl | Alt  | GUI  |Lower |Space |Space |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_DVORAK] = LAYOUT_48(
```
```suggestion
[_DVORAK] = LAYOUT_ortho_4x12(
```

zvecr commented on keyboards/montsinger/rebound/keymaps/default48/keymap.c:
```diff
@@ -0,0 +1,163 @@
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_48(
```
```suggestion
[_LOWER] = LAYOUT_ortho_4x12(
```

zvecr commented on keyboards/montsinger/rebound/keymaps/default48/keymap.c:
```diff
@@ -0,0 +1,163 @@
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_48(
```
```suggestion
[_RAISE] = LAYOUT_ortho_4x12(
```

zvecr commented on keyboards/montsinger/rebound/keymaps/default48/keymap.c:
```diff
@@ -0,0 +1,163 @@
+ * |      |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_ADJUST] =  LAYOUT_48(
```
```suggestion
[_ADJUST] =  LAYOUT_ortho_4x12(
```

Rossman360 commented:
I think I've made the suggested changes, but I get errors compiling the default48 keymap because of the implicit declaration of the community layout, when it wasn't doing that when I had named given it a custom name

zvecr commented on keyboards/montsinger/rebound/rebound.h:
```diff
@@ -26,7 +26,7 @@
  */
  
-#define LAYOUT_ortho_4x12( \
+#define LAYOUT_ortho4x12( \
```
```suggestion
#define LAYOUT_ortho_4x12( \
```

Rossman360 replied:
Ah, yes, my friend Underscore, glad to see y'all've met. Thanks for the help bud!

noroadsleft commented on keyboards/montsinger/rebound/readme.md:
```diff
@@ -5,11 +5,13 @@
 
 Make example for this keyboard (after setting up your build environment):
 
     make montsinger/rebound:default
```
As currently implemented, the make command must specify either `/rev1` or `/rev2`; this syntax won't work.

```suggestion
```

Rossman360 replied:
the change for line 8 is how my file already looked

drashna commented on keyboards/montsinger/rebound/rev2/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,162 @@
+
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_QWERTY);
+      }
+      return false;
+    case COLEMAK:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_COLEMAK);
+      }
+      return false;
+    case DVORAK:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_DVORAK);
+      }
+      return false;
+  }
+  return true;
+}
```
Just a couple of nit pick, but spacing:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
    return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
}

bool process_record_user(uint16_t keycode, keyrecord_t *record) {
    switch (keycode) {
        case QWERTY:
            if (record->event.pressed) {
                set_single_persistent_default_layer(_QWERTY);
             }
             break;
        case COLEMAK:
            if (record->event.pressed) {
                set_single_persistent_default_layer(_COLEMAK);
            }
            break;
        case DVORAK:
            if (record->event.pressed) {
                set_single_persistent_default_layer(_DVORAK);
            }
            break;
    }
    return true;
}
```

Rossman360 commented:
anything needed from me at this time?

Erovia commented:
You could also create a keyboard level `rules.mk` and use the `DEFAULT_FOLDER` directive. With this the `rebound` build target will not brake for users.
You can check the lets_split files for more detail.

noroadsleft commented on keyboards/montsinger/rebound/rules.mk:
```diff
@@ -1,32 +1 @@
-
-LAYOUTS = ortho_4x12
-LAYOUTS_HAS_RGB=no
+DEFAULT_FOLDER = montsinger/rebound/rev2
```
IMO this should point at rev1, which maintains behaviour for the existing users.

```suggestion
DEFAULT_FOLDER = montsinger/rebound/rev1
```

zvecr commented:
Thanks!
